libs/data-mapper/src/lib/mapDefinitions/MapDefinitionDeserializer.ts

```diff
@@ -192,19 +192,37 @@ const createConnections = (
   targetSchemaFlattened: SchemaNodeDictionary,
   functions: FunctionData[]
 ) => {
+  const isLoop: boolean = targetKey.includes(mapNodeParams.for);
+  const isConditional: boolean = targetKey.startsWith(mapNodeParams.if);
   const sourceEndOfFunction = sourceNodeString.indexOf('(');
-  const amendedSourceKey = targetKey.includes(mapNodeParams.for) ? getSourceValueFromLoop(sourceNodeString, targetKey) : sourceNodeString;
+  const amendedSourceKey = isLoop ? getSourceValueFromLoop(sourceNodeString, targetKey) : sourceNodeString;
+  let mockDirectAccessFnKey: string | undefined = undefined;
+  const [daOpenBracketIdx, daClosedBracketIdx] = [amendedSourceKey.indexOf('['), amendedSourceKey.indexOf(']')];
 
-  const sourceNode =
-    sourceEndOfFunction > -1
-      ? findFunctionForFunctionName(amendedSourceKey.substring(0, sourceEndOfFunction), functions)
-      : findNodeForKey(amendedSourceKey, sourceSchema.schemaTreeRoot);
+  // Identify source schema node, or Function(Data) from source key
+  let sourceNode: SchemaNodeExtended | FunctionData | undefined = undefined;
+  if (sourceEndOfFunction >= 0) {
+    // We found a Function in source key -> let's find its data
+    sourceNode = findFunctionForFunctionName(amendedSourceKey.substring(0, sourceEndOfFunction), functions);
+  } else if (daOpenBracketIdx >= 0 && daClosedBracketIdx >= 0) {
+    // One of the source key's chunks contained a Direct Access, so let's format it
+    // into the Function syntax the deserializer can parse
+    sourceNode = directAccessPseudoFunction;
+
+    mockDirectAccessFnKey = `directAccess(`;
```

Comment:
Callout: Could obviously include "directAccess(" and ")" in the inner string literals, but I personally prefer them as is to sort of clearly visually mark this string-building block